### PR TITLE
[HW4] Testing a Node.js Application with Jest

### PR DESCRIPTION
**Pull request by yevheniia-balda**
Test coverage report: 
image

**Requirements**
1.Test Setup:
- Set up a testing environment using Jest to write and run unit tests.
- Ensure that your project's dependencies include Jest and related testing libraries.

2.Unit Tests for APIs:
- Write unit tests for the API endpoints related to "Movies" and "Genres."
- Cover scenarios such as creating, reading, updating, and deleting movies and genres.
- Include tests for error handling and validation of input data.

3.Unit Tests for Movie Genre Search:
- Write unit tests for the movie genre search functionality.
- Test different genres and ensure that the API returns the expected results.
- Include edge cases and error scenarios in your tests.

4.Coverage and Reporting:
- Aim for high code coverage in your tests to ensure thorough testing of your application.
- Generate and include a code coverage report as part of your test results.

5.Documentation:
- Update your project's documentation to include information on running the tests using Jest.
- Provide clear instructions on how to run the tests and interpret the test results.

### CODE REVIEW

File: src/controllers/genres.controller.spec.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+      mockResponse,
+    );
+
+    expect(actualResult).toBe(expectedResult);
+  });
+
+  it("should respond with message when genre is not found on delete", async () => {
+    const mockRequest = {
+      params: { id: "nonExistentId" },
+    } as unknown as Request;
```
Comment (IhorOsadchyi) [inline src/controllers/genres.controller.spec.ts]: As I see you have used a similar mocks several times. You could move such mocks outside of a single test case and reuse it for less code repetition.

File: src/controllers/movies.controller.spec.ts [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        releaseDate: "2024-07-20",
+        genre: ["comedy"],
+      },
+    } as unknown as Request;
+    const expectedResult = {
+      _id: "mockId",
+      title: "Update Movie",
+      description: "description",
+      releaseDate: "2024-07-20",
+      genre: ["comedy"],
```
Comment (IhorOsadchyi) [inline src/controllers/movies.controller.spec.ts]: Same as above. This mocks are very similar. The content of the test fields is not essential here. You can create a universal mock and reuse it.

Comment (IhorOsadchyi) [reply src/controllers/movies.controller.spec.ts]: Please check for similar points throughout your code.

File: src/controllers/genres.controller.spec.ts (hunk 1 above)
Comment (yevheniia-balda) [reply src/controllers/genres.controller.spec.ts]: Reviewed the tests and moved common mock data outside of test cases.

File: src/controllers/movies.controller.spec.ts (hunk 2 above)
Comment (yevheniia-balda) [reply src/controllers/movies.controller.spec.ts]: Reviewed the tests and moved common mock data outside of test cases.

**Review by IhorOsadchyi — APPROVED** [review]
